lib/GADS/Role/Purgable.pm

```diff
@@ -0,0 +1,29 @@
+package GADS::Role::Purgable;
+
+use strict;
+use warnings;
+
+use Moo::Role;
+
+has valuefield => (
+    is      => 'lazy',
+    builder => sub { ('value'); }
```

Comment:
I'm not sure that arrays work reliably in Moo, I think better to use an arrayref, and then also set `isa` to `ArrayRef`